ddcz/views/legacy.py

```diff
@@ -0,0 +1,136 @@
+    "faq": "ddcz:website-manual",
+    "oao": "ddcz:faq",
+    "cojetodrd": "ddcz:about-drd",
+}
+
+LEGACY_LEGACY_CREATION_ROUTER = {
+    "hrbitov": "hrbitov",
+    "clanky": "clanky",
+    "expanze": "expanze",
+    "kouzelnikzvl": "kouzelnikzvl",
+    "alchymistazvl": "alchymistazvl",
+    "valecnik": "valecnik",
+    "hranicar": "hranicar",
+    "zlodej": "zlodej",
+    "novapovolani": "novapovolani",
+    "noverasy": "noverasy",
+}
+
+
+LEGACY_CREATION_ROUTER = [
+    "bestiar",
+    "dobrodruzstvi",
+    "predmety",
+    "kouzla",
+    "alchpredmety",
+    "hranicarkouzla",
+    "dovednosti",
+    "galerie",
+    "fotogalerie",
+]
+
+
+@require_http_methods(["GET"])
+def legacy_router(request):
+
+    get = dict(request.GET)
+    section = handle_get_key(get, "rub")
+    subsection = handle_get_key(get, "co")
+    id = handle_get_key(get, "id")
+
+    # The LEGACY_PLAIN_ROUTER is redirecting basic pages.
+    # Typically no creative pages are present here.
+    if section in LEGACY_PLAIN_ROUTER.keys():
+        return HttpResponseRedirect(reverse(LEGACY_PLAIN_ROUTER[section]))
+
+    # Some of the creation have legacy url
+    # index.php?rub=prispevky(_komp)&co=page_slug
+    # Here are lists of such creative pages.
+    if section in CREATION and subsection in LEGACY_LEGACY_CREATION_ROUTER.keys():
+        return HttpResponseRedirect(
+            reverse(
+                "ddcz:creation-list",
+                kwargs={
+                    "creative_page_slug": LEGACY_LEGACY_CREATION_ROUTER[subsection]
+                },
+            )
+        )
+
+    # For index.php?rub=prispevky_jeden&subsection=page_slug&id=article_id
+    # we can find the article by Id. Problem would be, if the ID
+    # if articles can be changed, than this will lead to the new
+    # article. There is probably no way how to pass by this. But
+    # changing the IDs of the articles is probably something no one
+    # will want to do.
+    if section == CREATION_DETAIL and id is not False:
+        page = get_object_or_404(CreativePage, slug=subsection)
+        return get_detail_redirect_response(page, id)
+
+    # There are some special creative pages that are not stored
+    # as the others, those have their own tables in the database.
+    # For those we have lists and details in this for loop.
+    for name in LEGACY_CREATION_ROUTER:
+        if section in [name, name + "_komp"]:
+            return HttpResponseRedirect(
+                reverse(
+                    "ddcz:creation-list",
+                    kwargs={"creative_page_slug": name},
+                )
+            )
+        if section in [name + "_jeden"]:
+            page = get_object_or_404(CreativePage, slug=name)
+            return get_detail_redirect_response(page, id)
+
+    ###  Finally if no route is found, redirect to news feed
+    return HttpResponseRedirect(reverse("ddcz:news"))
+
+
+def handle_get_key(get, key):
+    try:
+        return get[key][0]
+    except:
```

Comment:
`except KeyError`

Reply:
I had this before, but then there is a possibility of not returning anything, so I added `return False` at the end, but then again, this will catch anything and return it, seems more safe to me this way, is it not?